js/script.js

```diff
@@ -0,0 +1,116 @@
+$(document).ready(function() {
+
+	var images = [];
+
+	var counter = 0;
+
+
+	
+
+	$('button').on('click', function(e) { 
+		var $input = $("#input").val();
+		// console.log(input)
+		e.preventDefault();
+		// alert("I work");
+		$.get("https://www.reddit.com/search.json?q", {
+		q: $input
+
+	}).done(function(data) {
+		//console.log(data)
+		images = data.data.children;
+		// console.log(images)
+
+		function filterByUrl(obj) {
+			if (obj.data.url) {
```

Comment:
The reason you were getting some blanks is because you were using the ```obj.data.url``` value instead of the ```obj.data.preview.images``` value. Since all reddit posts have a url, you'll get a url, but it won't necessarily be a url to an image file, hence the blanks. I would do something like this instead:
```
if (obj.data.preview) {
```